--- configure.ac	2014-09-24 07:25:30.000000000 +0000
+++ configure.ac.new	2014-11-06 20:53:19.809918212 +0000
@@ -531,24 +531,9 @@
 AC_SUBST(GUDEV_CFLAGS)
 AC_SUBST(GUDEV_LIBS)
 
-# Make libv4l2 non-automagic
-AC_ARG_WITH([libv4l2],
-  AS_HELP_STRING([--with-libv4l2],[support video buffer conversion using libv4l2]),
-  [],
-  [with_libv4l2=check])
-if test x$HAVE_GST_V4L2 = xyes; then
-  if test x$with_libv4l2 != xno; then
-      PKG_CHECK_MODULES(LIBV4L2, libv4l2,
-          [ have_libv4l2=yes
-            AC_DEFINE(HAVE_LIBV4L2, 1,
-	        [Whether libv4l2 is available for video buffer conversion])
-          ], [
-          have_libv4l2=no
-          ])
-  else
-    have_libv4l2=no
-  fi
-fi
+
+have_libv4l2=no
+
 
 dnl Check for X11
 translit(dnm, m, l) AM_CONDITIONAL(USE_X, true)
